sofar/sofar.py

```diff
@@ -0,0 +1,2125 @@
+
+    Parameters
+    ----------
+    filename : str
+        The filename. '.sofa' is appended to the filename, if it is not
+        explicitly given.
+    sofa : object
+        The SOFA object that is written to disk
+    version : str
+        The SOFA object is verified and updated with :py:func:`~Sofa.verify`
+        before writing to disk. Version specifies, which version of the
+        convention is used:
+
+        ``'latest'``
+            Use the latest version upgrade the SOFA file if required.
+        ``'match'``
+            Match the version of the SOFA object.
+        str
+            Version string, e.g., ``'1.0'``.
+
+        The default is ``'latest'``.
+    compression : int
+        The level of compression with ``0`` being no compression and ``9``
+        being the best compression. The default of ``9`` optimizes the file
+        size but increases the time for writing files to disk.
+    """
+
+    # check the filename
+    if not filename.lower().endswith('.sofa'):
+        filename += ".sofa"
+
+    # setting the netCDF compression parameter
+    zlib = False if compression == 0 else True
+
+    # update the dimensions
+    sofa.verify(version)
+
+    # list of all attribute names
+    all_keys = [key for key in sofa.__dict__.keys() if not key.startswith("_")]
+
+    # open new NETCDF4 file for writing
+    with Dataset(filename, "w", format="NETCDF4") as file:
+
+        # write dimensions
+        for dim in sofa._api:
+            file.createDimension(dim, sofa._api[dim])
+
+        # write global attributes
+        keys = [key for key in all_keys if key.startswith("GLOBAL_")]
+        for key in keys:
+            setattr(file, key[7:], str(getattr(sofa, key)))
+
+        # write data
+        for key in all_keys:
+
+            # skip attributes
+            if ("_" in key and not key.startswith("Data_")) \
+                    or key.count("_") > 1:
+                continue
+
+            # get the data and type and shape
+            value, dtype = _format_value_for_netcdf(
+                getattr(sofa, key), key, sofa._convention[key]["type"],
+                sofa._dimensions[key], sofa._api["S"])
+
+            # create variable and write data
+            shape = tuple([dim for dim in sofa._dimensions[key]])
+            tmp_var = file.createVariable(
+                key.replace("Data_", "Data."), dtype, shape,
+                zlib=zlib, complevel=compression)
+            if dtype == "f8":
+                tmp_var[:] = value
+            else:
+                tmp_var[:] = stringtochar(value)
+                tmp_var._Encoding = "ascii"
+
+            # write variable attributes
+            sub_keys = [k for k in all_keys if k.startswith(key + "_")]
+            for sub_key in sub_keys:
+                setattr(tmp_var, sub_key[len(key)+1:],
+                        str(getattr(sofa, sub_key)))
+
+
+def compare_sofa(sofa_a, sofa_b, verbose=True, exclude=None):
```

Comment:
compare sofa indicates that the result of this function is the difference between both sofas. I'd rather name this to something like `check_equal`, or similar

Reply:
renamed it to `equals`